contributed_definitions/NXregion.nxdl.xml

```diff
@@ -0,0 +1,191 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<?xml-stylesheet type="text/xsl" href="nxdlformat.xsl" ?>
+<!--
+# NeXus - Neutron and X-ray Common Data Format
+# 
+# Copyright (C) 2013-2021 NeXus International Advisory Committee (NIAC)
+# 
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 3 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+#
+# For further information, see http://www.nexusformat.org
+-->
+<definition 
+	category="base" 
+	xmlns="http://definition.nexusformat.org/nxdl/3.1"
+	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	xsi:schemaLocation="http://definition.nexusformat.org/nxdl/3.1 ../nxdl.xsd"
+	name="NXregion" 
+	type="group" extends="NXobject">
+	<symbols>
+		<doc>
+			These symbols will denote how the shape of the parent group's data field,
+			
+			.. math:: (D_0, ..., D_{\mathbf{O}-1}, d_0, ..., d_{\mathbf{R}-1})
+			
+			could be split into a left set of **O** outer dimensions, :math:`\boldsymbol{D}`,
+			and a right set of **R** region dimensions, :math:`\boldsymbol{d}`,
+			where the data field has rank **O** + **R**. Note **O** :math:`>= 0`.
+		</doc>
+		<symbol name="O"><doc>Outer rank</doc></symbol>
+		<symbol name="R"><doc>Region rank</doc></symbol>
+	</symbols>
+	<doc>
+		Geometry and logical description of a region of data in a parent group. When used, it could
+		be a child group to, say, :ref:`NXdetector`.
+
+		This can be used to describe a subset of data used to create downsampled data or to derive
+		some data from that subset.
+
+		Note, the fields for the rectangular region specifiers follow HDF5’s dataspace hyperslab parameters
+		(see https://portal.hdfgroup.org/display/HDF5/H5S_SELECT_HYPERSLAB). Note when **block** :math:`= 1`,
```

Comment:
HDF5 hyperslab slicing is complicated. Isn't the [python slicing (`start`, `stop`, `step`)](https://docs.python.org/3/library/functions.html#slice) enough?

Reply:
No, we need `block` as well for downsampling